### PR TITLE
feat: spike to use Keycloak API instead of LDAP to retrieve users and groups

### PR DESCRIPTION
**Pull request by edgarvonk**
A spike to use Keycloak API instead of LDAP to retrieve users and groups. **DRAFT PR ONLY - WILL NOT BE MERGED**

Solves PZ-4837

### CODE REVIEW

File: .env.tpl [hunk 1]
```diff
@@ -30,14 +30,12 @@ FEATURE_FLAG_BPMN_SUPPORT=false
 GEMEENTE_CODE=password
 GEMEENTE_MAIL=op://Dimpact/ZAC-.env-$APP_ENV/GEMEENTE/MAIL
 GEMEENTE_NAAM=DummyGemeente
+KEYCLOAK_ADMIN_CLIENT_ID=zaakafhandelcomponent-admin-client
+KEYCLOAK_ADMIN_CLIENT_SECRET=zaakafhandelcomponentAdminClientSecret
```
Comment (xiduzo) [inline .env.tpl:34]: should this also go into the `.env.example`?

File: scripts/docker-compose/imports/keycloak/realm.json [hunk 2]
```diff
@@ -944,6 +944,109 @@
...
+      "enabled": true,
+      "alwaysDisplayInConsole": false,
+      "clientAuthenticatorType": "client-secret",
+      "secret": "zaakafhandelcomponentAdminClientSecret",
+      "redirectUris": [
+        "/*"
+      ],
+      "webOrigins": [
+        "/*"
+      ],
```
Comment (xiduzo) [inline scripts/docker-compose/imports/keycloak/realm.json:965]: can we make this stricter? or is this only used for running it locally in docker?

File: src/main/kotlin/net/atos/zac/identity/IdentityService.kt [hunk 3]
```diff
@@ -4,197 +4,58 @@
...
+    fun listGroups(): List<Group> = keycloakZacRealmResource.groups()
+        // retrieve groups with 'full representation' or else the group attributes will not be filled
+        .groups("", 0, Integer.MAX_VALUE, false)
+        .map { it.toGroup() }
+        .sortedBy { it.name }
+
+    fun readUser(userId: String): User = keycloakZacRealmResource.users()
+        .search(userId).map { it.toUser() }.firstOrNull()
+        // is this fallback really needed? better to return null or throw a custom exception
+        ?: User(userId)
```
Comment (xiduzo) [inline src/main/kotlin/net/atos/zac/identity/IdentityService.kt:41]: i would also throw an error or return null here as the user is not found

Comment (xiduzo) [reply src/main/kotlin/net/atos/zac/identity/IdentityService.kt:41]: similar to the other places where you do the same

File: .env.tpl (hunk 1 above)
Comment (edgarvonk) [reply .env.tpl:34]: Sure. This is just a spike so I did not finish the whole thing. :-)

File: scripts/docker-compose/imports/keycloak/realm.json (hunk 2 above)
Comment (edgarvonk) [reply scripts/docker-compose/imports/keycloak/realm.json:965]: Only for running locally in Docker. Also, it's just a spike.

File: src/main/kotlin/net/atos/zac/identity/IdentityService.kt (hunk 3 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/identity/IdentityService.kt:41]: yep. that's why I have added this todo. I suggest to look into this when we implement this for real. This fallback (which is also present in the current code) is really weird if you ask me.

**Pull request closed without merging**